crates/matrix-sdk/src/sliding_sync/builder.rs

```diff
@@ -296,7 +297,7 @@ impl SlidingSyncBuilder {
             rooms,
 
             position: Arc::new(AsyncMutex::new(SlidingSyncPositionMarkers { pos, delta_token })),
-            past_positions: StdRwLock::new(RingBuffer::new(20)),
+            past_positions: StdRwLock::new(RingBuffer::new(NonZeroUsize::new(20).unwrap())),
```

Comment:
(I see you've put SAFETY comments everywhere else but here, so in just you'd want to add it here, here's a review comment about it :-))